src/services/inlayHints.ts

```diff
@@ -0,0 +1,291 @@
+            if (enumValue !== undefined) {
+                addEnumMemberValueHints(enumValue.toString(), member.end);
+            }
+        }
+
+        function isModuleReferenceType(type: Type) {
+            return type.symbol && (type.symbol.flags & SymbolFlags.Module);
+        }
+
+        function visitVariableLikeDeclaration(decl: VariableDeclaration | PropertyDeclaration) {
+            const effectiveTypeAnnotation = getEffectiveTypeAnnotationNode(decl);
+            if (effectiveTypeAnnotation || !decl.initializer) {
+                return;
+            }
+
+            const declarationType = checker.getTypeAtLocation(decl);
+            if (isModuleReferenceType(declarationType)) {
+                return;
+            }
+
+            const typeDisplayString = printTypeInSingleLine(declarationType);
+            if (typeDisplayString) {
+                addTypeHints(typeDisplayString, decl.name.end);
+            }
+        }
+
+        function visitCallOrNewExpression(expr: CallExpression | NewExpression) {
+            const args = expr.arguments;
+            if (!args || !args.length) {
+                return;
+            }
+
+            const candidates: Signature[] = [];
+            const signature = checker.getResolvedSignatureForSignatureHelp(expr, candidates);
+            if (!signature || !candidates.length) {
+                return;
+            }
+
+            for (let i = 0; i < args.length; ++i) {
+                const arg = args[i];
+                if (!preferences.includeInlayNonLiteralParameterNameHints && !isHintableExpression(arg)) {
+                    continue;
+                }
+
+                const identifierNameInfo = checker.getParameterIdentifierNameAtPosition(signature, i);
+                if (identifierNameInfo) {
+                    const [parameterName, isFirstVariadicArgument] = identifierNameInfo;
+                    const isParameterNameNotSameAsArgument = preferences.includeInlayDuplicatedParameterNameHints || !isIdentifier(arg) || arg.text !== parameterName;
+                    if (!isParameterNameNotSameAsArgument && !isFirstVariadicArgument) {
+                        continue;
+                    }
+
+                    const name = unescapeLeadingUnderscores(parameterName);
+                    const commentRanges = getLeadingCommentRanges(sourceFileText, arg.pos);
+                    if (leadingCommentsHasParameterName(commentRanges, name)) {
+                        continue;
+                    }
+
+                    addParameterHints(name, args[i].getStart(), isFirstVariadicArgument);
+                }
+            }
+        }
+
+        function leadingCommentsHasParameterName(commentRanges: CommentRange[] | undefined, name: string) {
+            if (!commentRanges?.length) return false;
+            return commentRanges.some(range => stringContains(sourceFileText.substring(range.pos, range.end), name));

```

Comment:
I don't know if this is exactly the logic I'd recommend. My gut says it's something like the regex

```
^\s*(originalString)\s*:\s*$
```

Reply:
Not pretty sure but I guess I got your point.